bootstrap.py

```diff
@@ -1053,12 +1066,78 @@ def exec_service(service, command, failonerror=True):
+            hostgroup_id = return_matching_foreman_key('hostgroups', 'title="%s"' % options.hostgroup, 'id', False)
+            jdata = json.loads('{"host": {"hostgroup_id": "%s"}}' % hostgroup_id)
+            put_json("https://" + options.foreman_fqdn + ":" + API_PORT + "/api/hosts/%s" % host_id, jdata)
+        if options.location:
+            print_running("Calling Foreman API to switch location for %s to %s" % (FQDN, options.location))
+            location_id = return_matching_foreman_key('locations', 'title="%s"' % options.location, 'id', False)
+            jdata = json.loads('{"host": {"location_id": "%s"}}' % location_id)
+            put_json("https://" + options.foreman_fqdn + ":" + API_PORT + "/api/hosts/%s" % host_id, jdata)
+
+        # Configure new proxy_id for Puppet (if not skipped), and OpenSCAP (if available and not skipped)
+        if 'foreman' not in options.skip and 'puppet' not in options.skip:
+            print_running("Calling Foreman API to update Puppet master and Puppet CA for %s to %s" % (FQDN, options.foreman_fqdn))
+            update_host_capsule_mapping("puppet_proxy_id", capsule_id, host_id)
+            update_host_capsule_mapping("puppet_ca_proxy_id", capsule_id, host_id)
+        if 'foreman' not in options.skip and 'Openscap' in capsule_features:
+            print_running("Calling Foreman API to update OpenSCAP proxy for %s to %s" % (FQDN, options.foreman_fqdn))
+            update_host_capsule_mapping("openscap_proxy_id", capsule_id, host_id)
+        elif 'foreman' not in options.skip and 'Openscap' not in capsule_features:
+            print_warning("New capsule doesn't have OpenSCAP capability, not switching / configuring openscap_proxy_id")
+
+        print_running("Calling Foreman API to update content source for %s to %s" % (FQDN, options.foreman_fqdn))
+        update_host_capsule_mapping("content_source_id", capsule_id, host_id)
+
+        enable_rhsmcertd()
+
+        if 'puppet' not in options.skip and 'foreman' not in options.skip:
+            print_running("Stopping the Puppet agent for configuration update")
+            exec_service("puppet", "stop")
+
+            # Not using clean_puppet() and install_puppet_agent() here, because
+            # that would nuke custom /etc/puppet/puppet.conf files, which might
+            # yield undesirable results.
+            print_running("Updating Puppet configuration")
+            exec_failexit("sed -i 's/^[[:space:]]*server.*/   server     = %s/' /etc/puppet/puppet.conf" % options.foreman_fqdn)
+            exec_failok("sed -i 's/^[[:space:]]*ca_server.*/   server     = %s/' /etc/puppet/puppet.conf" % options.foreman_fqdn)  # For RHEL5 stock puppet.conf
+            delete_directory("/var/lib/puppet/ssl")
+            delete_file("/var/lib/puppet/client_data/catalog/%s.json" % FQDN)
+
+            print_generic("Running Puppet in noop mode to generate SSL certs")
```

Comment:
that duplicates code from `install_puppet_agent`, while we can't use the function itself, shall we split out the "run noop and tell user to sign" part in an own function and call that it both places?

Reply:
Yeah, that sounds like a plan. Will look into it.